NEXTCLOUD.cmake

```diff
@@ -15,6 +15,66 @@ set( APPLICATION_VIRTUALFILE_SUFFIX "nextcloud" CACHE STRING "Virtual file suffi
 set( LINUX_PACKAGE_SHORTNAME "nextcloud" )
 set( LINUX_APPLICATION_ID "${APPLICATION_REV_DOMAIN}.${LINUX_PACKAGE_SHORTNAME}")
 
+option(VERBOSE "If enabled, more messages will appear in the output." OFF)
+option(DAILY_BUILD "If enabled, name variables will be appended with \"daily\" in order to reduce conflicts." OFF)
+option(TEST_BUILD "If enabled, name variables will be appended with \"test\" in order to reduce conflicts." ON)
+# message(STATUS "DAILY_BUILD is set to ${DAILY_BUILD}, and TEST_BUILD is set to ${TEST_BUILD}.")
+
+function(ADD_SUFFIXES SUFFIX_STRING SUFFIX_NAME)
+    string(TOLOWER ${SUFFIX_STRING} SUFFIX_LOWER)
+    string(APPEND APPLICATION_NAME " (${SUFFIX_STRING} Build)")
+    set(APPLICATION_NAME ${APPLICATION_NAME} PARENT_SCOPE)
+    string(APPEND APPLICATION_SHORTNAME "${SUFFIX_STRING}")
+    set(APPLICATION_SHORTNAME ${APPLICATION_SHORTNAME} PARENT_SCOPE)
+    string(APPEND APPLICATION_EXECUTABLE "${SUFFIX_LOWER}")
+    set(APPLICATION_EXECUTABLE ${APPLICATION_EXECUTABLE} PARENT_SCOPE)
+    string(APPEND APPLICATION_ICON_NAME "${SUFFIX_STRING}")
+    set(APPLICATION_ICON_NAME ${APPLICATION_ICON_NAME} PARENT_SCOPE)
+    string(APPEND APPLICATION_REV_DOMAIN "${SUFFIX_LOWER}")
+    set(APPLICATION_REV_DOMAIN ${APPLICATION_REV_DOMAIN} PARENT_SCOPE)
+    string(APPEND LINUX_PACKAGE_SHORTNAME "${SUFFIX_LOWER}")
+    set(LINUX_PACKAGE_SHORTNAME ${LINUX_PACKAGE_SHORTNAME} PARENT_SCOPE)
+    string(APPEND LINUX_APPLICATION_ID "${SUFFIX_LOWER}")
+    set(LINUX_APPLICATION_ID ${LINUX_APPLICATION_ID} PARENT_SCOPE)
+endfunction()
+
+function(PRINT_SUFFIXES SUFFIX_NAME)
+    message(STATUS "${SUFFIX_NAME} is set to ${${SUFFIX_NAME}}, and name variables will be appended to prevent conflicts.")
+    if(VERBOSE)
+        message(NOTICE "   Set ${SUFFIX_NAME} to False or OFF to disable the ${SUFFIX_NAME} suffix.")
+        message(NOTICE "")
+        message(NOTICE "   Your choice will be cached. For further explanation, see:")
+        message(NOTICE "   https://cmake.org/cmake/help/latest/command/option.html")
+        message(NOTICE "")
+        message(NOTICE " * APPLICATION_NAME           is now:   \"${APPLICATION_NAME}\"")
+        message(NOTICE " * APPLICATION_SHORTNAME      is now:   \"${APPLICATION_SHORTNAME}\"")
+        message(NOTICE " * APPLICATION_EXECUTABLE     is now:   \"${APPLICATION_EXECUTABLE}\"")
+        message(NOTICE " * APPLICATION_ICON_NAME      is now:   \"${APPLICATION_ICON_NAME}\"")
+        message(NOTICE " * APPLICATION_REV_DOMAIN     is now:   \"${APPLICATION_REV_DOMAIN}\"")
+        message(NOTICE " * LINUX_PACKAGE_SHORTNAME    is now:   \"${LINUX_PACKAGE_SHORTNAME}\"")
+        message(NOTICE " * LINUX_APPLICATION_ID       is now:   \"${LINUX_APPLICATION_ID}\"")
+    endif()
+    message(NOTICE "")
+endfunction()
+
+if (DAILY_BUILD OR TEST_BUILD)
+    message(NOTICE "")
+endif()
```

Comment:
I mean, I could simplify it again. The main thing is that IMHO if `DEV_BUILD` is turned on, then it should at minimum print a status message to that effect, though rest of the output is largely unnecessary. I'm personally a fan of inline documentation and allowing people to opt into extremely verbose output, but if there's a better way of doing those things, feel free to suggest one. I do think that `DEV_BUILD` should be on by default, because it largely operates as a safety filter. Again, it does cache the user's choice, so it wouldn't need to be disabled on every run. (The reason I had initially tethered the output to the build type is that the existing build scripts would presumably work around the suffix by default.)

Reply:
> I mean, I could simplify it again. The main thing is that IMHO if DEV_BUILD is turned on, then it should at minimum print a status message to that effect, though rest of the output is largely unnecessary.

I'm fine with one message :) 
 
> I'm personally a fan of inline documentation and allowing people to opt into extremely verbose output, but if there's a better way of doing those things, feel free to suggest one.

In my opinion the option statement should be enough inline documentation. There is even a description string on the option. Extrem verbose output is in my opinion not a great idea in general. Just print the things that are really necessary. Everything else has the potential to confuse users, like you were in the beginning confused by all this warnings that CMake printed :) 

> I do think that DEV_BUILD should be on by default

I don't think so. A build without any options should in my opinion give you a release build that you can distribute. dev builds are a optional thing. CMake has become somewhat a standard in the C++ world. When people see a CMakeLists.txt in a project they already now that to get a proper build they just have to run
```
mkdir build
cd build
cmake ..
make
```
Because of that it's in my opinion not intuitiv to set a option to get a proper build + we would need to adjust the commands in our brander and other scripts.

Reply:
Yes, I agree with @FlexW statements here. I'm also a bit surprised this is so heavily cmake driven and require new icons. Why not colorize the existing one at build time or run time (my preference) for the couple of places it matters instead? I mean this seems to be mostly a "am I running the version I expect?" thing

Reply:
The full set of new icons was necessary IIRC because of the difference in the name (some of them could be aliases, though). I mean, it would be nice for the recoloring to be done programmatically, but it wasn't particularly straightforward how to do that. The orange is the same saturation and luminance, with the hue rotated 180º, while the pink is IIRC rotated 120º. (Orange for caution, and pink borrowed from [Firefox's](https://wiki.mozilla.org/Nightly) nightly icon.) If you have a recommendation of how I could more specifically simplify the icon recoloring, I'd be happy to incorporate it.